### PR TITLE
Ensure to use the profile or selectedAppLanguage when format the dates

### PR DESCRIPTION
**Pull request by framitdavid**
## Description

Using the user's profile or the selected app language to format dates accurately. Using the appropriate language setting is crucial to ensure that dates are displayed correctly in the user's desired format.

image

## Related Issue(s)

The PR itself.

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [ ] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [x] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/utils/dateHelpers.ts [hunk 1]
```diff
@@ -21,8 +21,11 @@ export function getISOString(potentialDate: string | undefined): string | undefi
 const locale = window.navigator?.language || (window.navigator as any)?.userLanguage || 'nb';
 moment.locale(locale);
 
-export function getDateFormat(format?: string): string {
-  return moment.localeData().longDateFormat('L') || format || DatepickerFormatDefault;
+export function getDateFormat(format?: string, selectedLanguage = 'nb'): string {
```
Comment (bjosttveit) [inline src/utils/dateHelpers.ts:24]: I see you used a default value for selectedLanguage. However, to solve the issue in PDF, it also needs to pass this parameter in the other places this function gets used.

- You need to also include it in `Datepicker/index.ts` -> `useDisplayData()` for summary and PDF to work.
- `validateDatepickerFormData()` in `validation.ts` to get correct validation messages.
- Would also be nice to fix `InstanceInformationComponent` to get correct formatting for the date the instance was sent.
- In `ReceiptContainer`:123 the wrong formatting is also used.

**Review by bjosttveit — CHANGES_REQUESTED** [review]
To fix the PDF issue you at least have to update `useDisplayData()` in the Datepicker component class. You don't have to do all of the other suggested changes, but those are also similar issues.

File: src/utils/dateHelpers.ts (hunk 1 above)
Comment (framitdavid) [reply src/utils/dateHelpers.ts:24]: Thanks for feedback. 👍

File: src/layout/Datepicker/index.tsx [hunk 2]
```diff
@@ -17,11 +17,14 @@ export class Datepicker extends FormComponent<'Datepicker'> {
 
   useDisplayData(node: LayoutNodeFromType<'Datepicker'>): string {
     const formData = useAppSelector((state) => state.formData.formData);
+    const language = useAppSelector(
+      (state) => state.profile.selectedAppLanguage || state.profile.profile.profileSettingPreference.language,
+    );
```
Comment (bjosttveit) [inline src/layout/Datepicker/index.tsx:22]: We have an `appLanguageStateSelector` that does this logic++

**Review by bjosttveit — APPROVED** [review]

**Pull request merged**